service/header/sync.go

```diff
@@ -81,17 +89,59 @@ func (s *Syncer) Start(ctx context.Context) error {
 }
 
 // Stop stops Syncer.
-func (s *Syncer) Stop(context.Context) error {
+func (s *Syncer) Stop(ctx context.Context) error {
+	err := s.WaitSync(ctx)
```

Comment:
> I guess you could say that in both the unhappy and happy case you have to do the same thing, but it still seems unidiomatic.

Yes, we need that code to happen for both happy and unhappy cases.  The other ways of doing this are:
* Copying the code for both branches(obviously no)
* Extracting those lines in between in a separate func and calling it from two code branches(no, as I don't want to create a separate func for those few lines that are already logically part of Stop func)
* Using defer
	* Using one defer that wraps those locs in an anonymous func(or separate one(no))
		```go
			defer func() {
				s.heightSub.Stop()
			    s.cancel()
			    s.cancel = nil
			}
		```	
	* This would make code a bit bulkier with two more lines(what i don't like), though it might increase readability
	* Another option would be to have multiple defers
		```go
			defer s.heightSub.Stop()
		  	defer s.cancel()
		 	defer s.cancel = nil
			return s.WaitSync(ctx)
		```
		* Unfortunately third line is not possible and it forces to have anonymous func defer
I still would prefer to save those two lines(personal bug to make this as succinct as possible), but I am not married to them and will change two use defer.